app/scripts/lib/marketing-email-client.js

```diff
@@ -0,0 +1,78 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/**
+ * A client to talk to the basket marketing email server
+ */
+
+'use strict';
+
+define([
+  'lib/xhr',
+  'lib/marketing-email-errors'
+], function (xhr, MarketingEmailErrors) {
+
+  function MarketingEmailClient(options) {
+    options = options || {};
+
+    var self = this;
+
+    self._xhr = options.xhr || xhr;
+    self._baseUrl = options.baseUrl;
+    self._preferencesUrl = options.preferencesUrl;
+  }
+
+  MarketingEmailClient.prototype = {
+    _request: function (path, type, accessToken, data) {
```

Comment:
In the [oauth-client](https://github.com/mozilla/fxa-content-server/pull/2452/files#diff-6ae1cc09e16392740c1449123b7c1787R24), the parameters are `method`, `endpoint`, `params`. Might as well go for consistency.